posthog/hogql_queries/insights/test/test_lifecycle_query_runner.py

```diff
@@ -36,6 +36,544 @@ def create_action(**kwargs):
     return action
 
 
+class TestLifecycleQueryRetentionGroupAggregation(ClickhouseTestMixin, APIBaseTest):
+    maxDiff = None
+
+    def _create_events(self, data, event="$pageview"):
+        person_result = []
+        for id, timestamps_and_groups in data:
+            with freeze_time(timestamps_and_groups[0][0]):
+                person_result.append(
+                    _create_person(
+                        team_id=self.team.pk,
+                        distinct_ids=[id],
+                        properties={
+                            "name": id,
+                            **({"email": "test@posthog.com"} if id == "p1" else {}),
+                        },
+                    )
+                )
+            for timestamp, groups in timestamps_and_groups:
+                # not sure this is right, this maybe should put groups in directly?
```

Comment:
Possibly. Many other tests do this: https://github.com/PostHog/posthog/blob/5513be7731414cb1a4ffa5a7789610005a8538ae/ee/clickhouse/views/test/funnel/test_clickhouse_funnel.py#L27-L41